src/core/simulation.jl

```diff
@@ -110,6 +110,61 @@ function unregister_recorders!(internal::SimulationInternal)
     end
 end
 
+function _add_initial_condition_caches(
+    sim::SimulationInternal,
+    stage::Stage,
+    caches::Union{Nothing, Vector{<:AbstractCache}},
```

Comment:
Add a debug statement here is the caches are `nothing`

Reply:
Since the nothing will be removed that won't be needed